src/coreclr/jit/lclvars.cpp

```diff
@@ -2943,31 +2939,34 @@ void Compiler::lvaSetStruct(unsigned varNum, CORINFO_CLASS_HANDLE typeHnd, bool
 #endif // FEATURE_IMPLICIT_BYREFS
 
 #if FEATURE_SIMD
-            if (simdBaseJitType != CORINFO_TYPE_UNDEF)
+            if (structSizeMightRepresentSIMDType(layout->GetSize()))
```

Comment:
This could use `varTypeIsSIMD(layout->GetType())` as the fast path check (`impNormStructType` would still have to be called to get the base type).